projects/valtimo/zgw/src/lib/modules/zaakdetails/components/document-objecten-api-sync/document-objecten-api-sync.component.ts

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright 2015-2024 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Component, OnDestroy, OnInit} from '@angular/core';
+import {
+  AbstractControl,
+  FormControl,
+  FormGroup,
+  FormsModule,
+  ReactiveFormsModule,
+  Validators,
+} from '@angular/forms';
+import {ActivatedRoute} from '@angular/router';
+import {
+  CARBON_THEME,
+  FormModule,
+  InputModule,
+  SelectItem,
+  SelectModule,
+  SpinnerModule,
+  ValtimoCdsModalDirectiveModule,
+} from '@valtimo/components';
+import {TranslateModule} from '@ngx-translate/core';
+import {BehaviorSubject, map, Observable, Subscription, switchMap, tap} from 'rxjs';
+import {CommonModule} from '@angular/common';
+import {ButtonModule, CheckboxModule, ModalModule} from 'carbon-components-angular';
+import {DocumentObjectenApiSyncService} from '../../services';
+import {DocumentObjectenApiSync} from '../../models';
+import {DocumentDefinition} from '@valtimo/document';
+
+@Component({
+  selector: 'valtimo-document-objecten-api-sync',
+  templateUrl: './document-objecten-api-sync.component.html',
+  styleUrls: ['./document-objecten-api-sync.component.scss'],
+  standalone: true,
+  imports: [
+    CommonModule,
+    FormsModule,
+    TranslateModule,
+    SpinnerModule,
+    FormModule,
+    InputModule,
+    SelectModule,
+    ValtimoCdsModalDirectiveModule,
+    CheckboxModule,
+    ReactiveFormsModule,
+    ButtonModule,
+    ModalModule,
+  ],
+})
+export class DocumentObjectenApiSyncComponent implements OnInit, OnDestroy {
+  readonly loading$ = new BehaviorSubject<boolean>(true);
```

Comment:
public readonly. Please add public/private to all class props where they are missing